lib/vsc/install/shared_setup.py

```diff
@@ -263,12 +263,29 @@ def _fvs(msg=None):
 
 
 def _read(source, read_lines=False):
-    """read a file, either in full or as a list (read_lines=True)"""
-    with open(source) as file_handle:
+    """read a file, either in full or as a list (read_lines=True) and decode UTF-8 files into ASCII"""
```

Comment:
since you don't know that the `UnidecodeError` below comes from UTF-8 encoding issues, i think you should give `open(fname, encoding="ascii", errors="surrogateescape")` a try and see if it solves your issues (or `encoding=latin-1`).

Reply:
AFAIK there is no other way that a `UnicodeError` can be triggered if it is not by the presence of unicode characters. So the `ascii` or `latin9` encodings won't fix the issue.

Reply:
but can you properly read `utf-16` text with a `utf-8` encoding? both opens abvoe will handle utf8 without errors

Reply:
Ok, it might be safer to directly decode the file in ASCII and handle the errors. Using `surrogateescape` works to read the contents of the file
```
open(source, encoding='ascii', errors='surrogateescape') as file_handle:
    txt = file_read(file_handle, read_lines)
```

but it generates new errors down the line. There is a write operation in UTF-8 which does not allow surrogates
```
INFO: writing lib/vsc_install.egg-info/PKG-INFO
Traceback (most recent call last):
  File "setup.py", line 1675, in <module>
    action_target(PACKAGE)
  File "setup.py", line 1642, in action_target
    _fvs('action_target function')().action_target(package, *args, **kwargs)
  File "setup.py", line 1631, in action_target
    setupfn(**x)
  File "/usr/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
[...]
  File "/usr/lib/python3.6/site-packages/setuptools/dist.py", line 92, in write_pkg_file
    file.write('Description: %s\n' % long_desc)
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 10383-10385: surrogates not allowed
error: Bad exit status from /var/tmp/rpm-tmp.zPj48a (%install)
```

I could hunt all those new errors, but I think that we can just use `backslashreplace`. This works across the board and the result is similar to `surrogateescape`. Done in https://github.com/hpcugent/vsc-install/pull/175/commits/22be9bc05d09829b8b8e37fc150d520d2eeede32